src/validations/css/stylelint.js

```diff
@@ -0,0 +1,65 @@
+import Validator from '../Validator';
+
+const errorMap = {
+  // CSS Syntax Errors
+  'Unclosed block': (error) => {
+    const token = error.source;
+
+    return {
+      reason: 'missing-closing-curly',
+      payload: {error: token},
+    };
+  },
+
+  // Custom Rules
+  'declaration-block-trailing-semicolon': (error) => {
+    const token = error.css;
+
+    return {
+      reason: 'missing-semicolon',
+      payload: {error: token},
+    };
+  },
+};
+
+class StyleLintValidator extends Validator {
+  constructor(source) {
+    super(source, 'css', errorMap);
+  }
+
+  _getRawErrors() {
+    return System.import('../linters').then(
+      ({stylelint}) => stylelint(
+        this._source
+      ).then(
+        (result) => ({result})
+      ).catch(
+        (syntaxError) => ({syntaxError})
+      )
```

Comment:
This is subtle, but in this situation it’s preferable to pass both success and error callbacks to `then`, like so:

```js
      stylelint(this._source).then(
        (result) => ({result}),
        (syntaxError) => ({syntaxError})
      )
```

The reason is that we don’t want our `catch` to catch rejections generated *in the `then` callback*—we only want to catch rejections from the initial call to `stylelint(this._source)`.